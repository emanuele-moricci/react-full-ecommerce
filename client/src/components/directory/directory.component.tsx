import { useSelector } from "react-redux";
import { selectSections } from "src/redux/directory/directory.selectors";

import DirectoryItem from "src/components/directory/directory-item/directory-item.component";

import * as Styled from "./directory.styles";

const Directory = (): JSX.Element => {
  const sections = useSelector(selectSections);

  return (
    <Styled.DirectoryMenu>
      {sections.map((section) => (
        <DirectoryItem key={section.id} {...section} />
      ))}
    </Styled.DirectoryMenu>
  );
};

export default Directory;
