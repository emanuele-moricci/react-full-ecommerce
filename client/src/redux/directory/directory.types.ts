export type Section = {
  title: string;
  imageUrl: string;
  linkUrl: string;
  size?: string;
  id: number;
};
