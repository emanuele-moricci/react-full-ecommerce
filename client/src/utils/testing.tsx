import { FC } from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { BrowserRouter } from "react-router-dom";

export const fakePayloadAction = { type: "FAKE_PAYLOAD" };

export const getMockedStoreRenderedComponent = (state: any, Component: FC) => {
  const mockStore = configureStore();
  const store = mockStore(state);

  return renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Component />
      </BrowserRouter>
    </Provider>
  );
};
