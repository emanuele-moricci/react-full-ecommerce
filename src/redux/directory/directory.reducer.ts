import { AnyAction } from "redux";

const INITIAL_STATE = {
  sections: [
    {
      title: "hats",
      imageUrl: "https://i.ibb.co/cvpntL1/hats.png",
      linkUrl: "hats",
      id: 1,
    },
    {
      title: "jackets",
      imageUrl: "https://i.ibb.co/px2tCc3/jackets.png",
      linkUrl: "jackets",
      id: 2,
    },
    {
      title: "sneakers",
      imageUrl: "https://i.ibb.co/0jqHpnp/sneakers.png",
      linkUrl: "sneakers",
      id: 3,
    },
    {
      title: "womens",
      imageUrl: "https://i.ibb.co/GCCdy8t/womens.png",
      linkUrl: "womens",
      size: "large",
      id: 4,
    },
    {
      title: "mens",
      imageUrl: "https://i.ibb.co/R70vBrQ/men.png",
      linkUrl: "mens",
      size: "large",
      id: 5,
    },
  ],
};

const directoryReducer = (state = INITIAL_STATE, { type }: AnyAction) => {
  switch (type) {
    default:
      return state;
  }
};

export default directoryReducer;
