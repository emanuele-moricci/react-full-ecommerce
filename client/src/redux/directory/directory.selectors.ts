import { createSelector } from "reselect";

import { RootState } from "src/redux/root.store";
import { Section } from "src/redux/directory/directory.types";

const selectDirectory = (state: RootState) => state.directory;

export const selectSections = createSelector(
  [selectDirectory],
  (directory): Section[] => directory.sections
);
