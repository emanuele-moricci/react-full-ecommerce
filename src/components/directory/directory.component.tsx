import { useState } from "react";

import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectSections } from "../../redux/directory/directory.selectors";
import { Section } from "../../redux/directory/directory.types";

import MenuItem from "../menu-item/menu-item.component";

import "./directory.styles.scss";

interface IDirectoryProps {
  sections: Section[];
}

const Directory = ({ sections }: IDirectoryProps): JSX.Element => {
  const [dirSections] = useState<Section[]>(sections);

  return (
    <div className="directory-menu">
      {dirSections.map((section) => (
        <MenuItem key={section.id} {...section} />
      ))}
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  sections: selectSections,
});

export default connect(mapStateToProps)(Directory);
