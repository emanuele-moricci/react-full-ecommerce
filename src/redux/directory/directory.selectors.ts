import { createSelector } from "reselect";

import { Section } from "./directory.types";

const selectDirectory = (state: any) => state.directory;

export const selectSections = createSelector(
  [selectDirectory],
  (directory): Section[] => directory.sections
);
