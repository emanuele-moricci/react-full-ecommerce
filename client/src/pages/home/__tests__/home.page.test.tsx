import { getMockedStoreRenderedComponent } from "src/utils/testing";

import { initialState } from "../state/directory.slice";

import Home from "src/pages/home/home.page";

describe("<Home />", () => {
  it("Should match the snapshot", () => {
    const component = getMockedStoreRenderedComponent(
      { directory: initialState },
      Home
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
