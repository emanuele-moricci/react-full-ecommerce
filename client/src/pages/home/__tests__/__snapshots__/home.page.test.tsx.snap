// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Should match the snapshot 1`] = `
<div
  className="sc-gKclnd fHrYJH"
>
  <div
    className="sc-jRQBWg jecsyG"
  >
    <div
      className="sc-bdvvtL cUwxXM"
      onClick={[Function]}
    >
      <div
        className="sc-gsDKAQ kjUoh background-image"
      />
      <div
        className="sc-dkPtRN iPlZZe content"
      >
        <span
          className="sc-hKwDye kMQOTw title"
        >
          HATS
        </span>
        <span
          className="sc-eCImPb joXWrW subtitle"
        >
          SHOP NOW
        </span>
      </div>
    </div>
    <div
      className="sc-bdvvtL cUwxXM"
      onClick={[Function]}
    >
      <div
        className="sc-gsDKAQ gDMcnR background-image"
      />
      <div
        className="sc-dkPtRN iPlZZe content"
      >
        <span
          className="sc-hKwDye kMQOTw title"
        >
          JACKETS
        </span>
        <span
          className="sc-eCImPb joXWrW subtitle"
        >
          SHOP NOW
        </span>
      </div>
    </div>
    <div
      className="sc-bdvvtL cUwxXM"
      onClick={[Function]}
    >
      <div
        className="sc-gsDKAQ dmMrFS background-image"
      />
      <div
        className="sc-dkPtRN iPlZZe content"
      >
        <span
          className="sc-hKwDye kMQOTw title"
        >
          SNEAKERS
        </span>
        <span
          className="sc-eCImPb joXWrW subtitle"
        >
          SHOP NOW
        </span>
      </div>
    </div>
    <div
      className="sc-bdvvtL dSnQlR"
      onClick={[Function]}
      size="large"
    >
      <div
        className="sc-gsDKAQ gASBNm background-image"
      />
      <div
        className="sc-dkPtRN iPlZZe content"
      >
        <span
          className="sc-hKwDye kMQOTw title"
        >
          WOMENS
        </span>
        <span
          className="sc-eCImPb joXWrW subtitle"
        >
          SHOP NOW
        </span>
      </div>
    </div>
    <div
      className="sc-bdvvtL dSnQlR"
      onClick={[Function]}
      size="large"
    >
      <div
        className="sc-gsDKAQ ibfNaN background-image"
      />
      <div
        className="sc-dkPtRN iPlZZe content"
      >
        <span
          className="sc-hKwDye kMQOTw title"
        >
          MENS
        </span>
        <span
          className="sc-eCImPb joXWrW subtitle"
        >
          SHOP NOW
        </span>
      </div>
    </div>
  </div>
</div>
`;
